src/cross_section.F90

```diff
@@ -526,6 +572,185 @@ pure function find_energy_index(mat, E) result(i)
+        end if
+
+        temp = temp * sqrtE
+      end do
+    end if
+
+    ! Then get the poles we want and broaden them.
+
+    if (sqrtkT == 0.0_8) then
+      ! If at 0K, use asymptotic form.
+      do iP = startw, endw
+        PSIIKI = -ONEI/(multipole % data(MP_EA, iP) - sqrtE)
+        CDUM1 = PSIIKI/E
+        if (multipole % formalism == FORM_MLBW) then
+          sigT = sigT + real(multipole % data(MLBW_RT, iP)*CDUM1* &
+                             sigT_factor(multipole % l_value(iP))) &
+                      + real(multipole % data(MLBW_RX, iP)*CDUM1)
+          sigA = sigA + real(multipole % data(MLBW_RA, iP)*CDUM1)
+          sigF = sigF + real(multipole % data(MLBW_RF, iP)*CDUM1)
+        else if (multipole % formalism == FORM_RM) then
+          sigT = sigT + real(multipole % data(RM_RT, iP)*CDUM1* &
+                             sigT_factor(multipole % l_value(iP)))
+          sigA = sigA + real(multipole % data(RM_RA, iP)*CDUM1)
+          sigF = sigF + real(multipole % data(RM_RF, iP)*CDUM1)
+        end if
+      end do
+    else
+      ! At temperature, use Faddeeva function-based form.
+      if(endw >= startw) then
+        do iP = startw, endw
+          Z = (sqrtE - multipole % data(MP_EA, iP))*DOPP
+          call w(Z, w_val)
+          w_val = w_val*DOPP_ECOEF
+
+          if (multipole % formalism == FORM_MLBW) then
+            sigT = sigT + real((multipole % data(MLBW_RT, iP)* &
+                          sigT_factor(multipole%l_value(iP)) + &
+                          multipole % data(MLBW_RX, iP))*w_val)
+            sigA = sigA + real(multipole % data(MLBW_RA, iP)*w_val)
+            sigF = sigF + real(multipole % data(MLBW_RF, iP)*w_val)
+          else if (multipole % formalism == FORM_RM) then
+            sigT = sigT + real(multipole % data(RM_RT, iP)*w_val* &
+                               sigT_factor(multipole % l_value(iP)))
+            sigA = sigA + real(multipole % data(RM_RA, iP)*w_val)
+            sigF = sigF + real(multipole % data(RM_RF, iP)*w_val)
+          end if
+        end do
+      end if
+    end if
+  end subroutine
+
+!===============================================================================
+! FILL_FACTORS calculates the value of phi, the hardsphere phase shift factor,
+! and sigT_factor, a factor inside of the sigT equation not present in the
+! sigA and sigF equations.
+!===============================================================================
+
+  subroutine fill_factors(multipole, sqrtE, sigT_factor, twophi, max_L)
+    type(MultipoleArray), intent(in)   :: multipole
+    real(8), intent(in)                  :: sqrtE
+    integer, intent(in)                  :: max_L
+    complex(8), intent(out)              :: sigT_factor(max_L)
+    real(8), intent(out)                 :: twophi(max_L)
+
+    integer :: iL
+    real(8) :: arg
+
+    do iL = 1, max_L
+      twophi(iL) = multipole%pseudo_k0RS(iL)*sqrtE
```

Comment:
Put spaces around the `%`, and consider spaces around math operators in this function (at your discretion)